src/toolbelt/core.cljc

```diff
@@ -151,20 +153,42 @@
   elements of xs return the same value under f, the first is returned"
   [f xs]
   (let [s (atom #{})]
-    (for [x     xs
-          :let  [id (f x)]
+    (for [x xs
+          :let [id (f x)]
           :when (not (contains? @s id))]
       (do (swap! s conj id)
           x))))
 
 
-(defn round
-  [x & [precision]]
-  (if (some? precision)
-    (let [scale (Math/pow 10 precision)]
-      (-> x (* scale) Math/round (/ scale)))
-    (Math/round x)))
+#?(:clj
+   (defn round
+     [x & [precision]]
+     (if (some? precision)
+       (let [scale (Math/pow 10 precision)]
+         (-> x (* scale) Math/round (/ scale)))
+       (Math/round x)))
+
+   :cljs
+   (defn round
```

Comment:
Do you prefer defining the functions separately over having the implementation be different? (I'm fine with both, I'm just curious)

Reply:
Me too :) 

Below Rich's code:
```
;; clojure.test.check.generators/choose

(defn choose
  #?(:clj
     "Create a generator that returns long integers in the range `lower` to `upper`, inclusive."

     :cljs
     "Create a generator that returns numbers in the range
     `lower` to `upper`, inclusive.")
  [lower upper]
  ;; cast to long to support doubles as arguments per TCHECK-73
  (core/let #?(:clj
          [lower (long lower)
           upper (long upper)]

          :cljs ;; does nothing, no long in cljs
          [])
    (make-gen
     (fn [rnd _size]
       (core/let [value (rand-range rnd lower upper)]
         (rose/filter
          #(and (>= % lower) (<= % upper))
          (int-rose-tree value)))))))```